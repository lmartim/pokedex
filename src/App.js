import React from 'react';
import { Switch, Route } from 'react-router-dom';
import { Columns } from 'react-bulma-components/dist';
import styled from "styled-components";

import FavoritesBar from './components/favorites-bar/favorites-bar.component';
import Home from './pages/home/home.component';
import Details from './pages/details/details.component';

import bg from './assets/images/pokeball-wallpaper.jpg';

// Configurado as rotas do projeto e seus respectivos componentes
// Também é adicionado o Header no topo da aplicação
function App() {
  return (
    <AppBlock>
      <AppBlockContent>
        <Columns>
          <Columns.Column size="one-third">
            <FavoritesBar />
          </Columns.Column>
          <Columns.Column size="two-thirds">
            <Switch>
              <Route exact path='/' component={Home} />
              <Route exact path='/detalhes' component={Details} />
            </Switch>
          </Columns.Column>
        </Columns>
      </AppBlockContent>
      <AppBlockBg bg={bg} />
    </AppBlock>
  );
}

export default App;

const AppBlock = styled.section `
  min-height: 100vh;
  position: relative;
  padding: 20px;
`

const AppBlockBg = styled.div `
  background-image: url(${props => props.bg});
  background-size: cover;
  width: 100vw;
  min-height: 100vh;
  height: 100%;
  top: 0;
  left: 0;
  position: absolute;
  z-index: -10;
  @media (max-width: 992px) {
    background-position: right;
  }
`

const AppBlockContent = styled.div `
  ::before {
    content: "";
    position: absolute;
    top: 0; 
    left: 0;
    border-style: solid;
    border-width: 100vh 40vh 0 0;
    border-color: #91C702 transparent transparent transparent;
    width: 40vw; 
    height: 100vh;
    z-index -1;
  }
  ::after {
    content: "";
    position: absolute;
    top: 0; 
    left: 0;
    border-style: solid;
    border-width: 90vh 40vh 0 0;
    border-color: #B4E637 transparent transparent transparent;
    width: 20vw; 
    height: 100vh;
    z-index -1;
  }
`;

