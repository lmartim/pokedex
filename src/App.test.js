import React from 'react';
import { render, cleanup } from './utils/test/test-utils';
import App from './App';

afterEach(cleanup)

// Testes relacionados ao componente App

// Renderiza o componente App e faz uma checagem por snapshot
// É necessário a atualizaçãod o snapshot, caso tenha mudança no componente
test('render app', () => {
  const { asFragment } = render(<App />)
    
  expect(asFragment(<App />)).toMatchSnapshot()
});
