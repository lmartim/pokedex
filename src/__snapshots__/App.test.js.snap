// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render app 1`] = `
<DocumentFragment>
  <section
    class="sc-fznxsB ecMVuZ"
  >
    <div
      class="sc-fznWqX inqrrU"
    >
      <div
        class="columns is-multiline"
      >
        <div
          class="column is-one-third"
        >
          <section
            class="sc-AxjAm StDqN"
          >
            <div
              class="sc-AxirZ gxLBLK"
              data-testid="title"
            >
              <b>
                FAVORITOS
              </b>
            </div>
            <div
              class="sc-AxirZ gxLBLK"
              data-testid="character"
            >
              <img
                alt="Pikachu"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png"
              />
              <div>
                Pikachu
              </div>
            </div>
          </section>
        </div>
        <div
          class="column is-two-thirds"
        >
          <section
            class="sc-fznyAO eOAqsL"
          >
            <div
              class="container"
            >
              <section
                class="sc-AxgMl jdza-dK"
                data-testid="pokemonList"
              >
                <div
                  class="sc-AxheI pyemq"
                >
                  <div
                    class="sc-Axmtr hUpDBo"
                  >
                    <div
                      class="columns is-multiline"
                    />
                  </div>
                </div>
                <section
                  class="sc-AxmLO kuQIeQ"
                >
                  <div
                    class="sc-fzozJi hOPdSQ"
                  >
                    Box #1
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-caret-right fa-w-6 "
                      data-icon="caret-right"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 192 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                  <div
                    class="sc-fzoLsD nSNYZ"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-search fa-w-16 "
                      data-icon="search"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                        fill="currentColor"
                      />
                    </svg>
                    <input
                      placeholder="Pesquisar"
                      type="text"
                    />
                  </div>
                </section>
              </section>
            </div>
          </section>
        </div>
      </div>
    </div>
    <div
      class="sc-fznJRM dgoZxw"
    />
  </section>
</DocumentFragment>
`;
